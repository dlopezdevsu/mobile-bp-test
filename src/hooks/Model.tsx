class model_class {

  dispatch: any = null
  get_dispatch = () => this.dispatch == null ? false : this.dispatch
  set_dispatch = (n_dispatch: any) => { this.dispatch = n_dispatch }

  setData = (label: string, payload: any) => {
    this?.dispatch({ type: 'SET_STORE', label, payload })
  }

  resetData = () => {
    this?.dispatch({ type: 'RESET_DATA' })
  }
}

export default new model_class();