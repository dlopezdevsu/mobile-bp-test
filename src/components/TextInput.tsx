import * as React from 'react';
import { TextInput, StyleSheet, TextInputProps, Text } from 'react-native';

import * as Colors from 'src/utils/colors';

interface TextInputComponentProps extends TextInputProps {
  label?: string;
  errorLabel?: string;
  hasError?: boolean;
}

const TextInputComponent = ({ label, errorLabel, hasError, editable, ...props }: TextInputComponentProps) => {

  return (
    <>
      <Text style={styles.label}>{label}</Text>
      <TextInput
        {...props}
        style={[styles.container, props.style, editable === false && styles.disabled, hasError && styles.inputError]}
        placeholderTextColor={Colors.black}
      />
      {hasError && <Text style={styles.error}>{errorLabel}</Text>}
    </>
  );
};

const styles = StyleSheet.create({
  label: {
    fontWeight: '600',
    fontSize: 12,
    marginBottom: 6,
    marginTop: 18
  },
  container: {
    borderWidth: 2,
    borderColor: Colors.gray,
    backgroundColor: Colors.white,
    padding: 12,
    borderRadius: 6
  },
  disabled: {
    backgroundColor: Colors.gray
  },
  error: {
    color: Colors.red,
    fontWeight: '600',
    fontSize: 12,
    marginTop: 6
  },
  inputError: {
    borderColor: Colors.red
  }
});

export default TextInputComponent;