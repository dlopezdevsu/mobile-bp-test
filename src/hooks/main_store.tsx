import React, { ReactElement } from "react";

import { ItemProductListProps } from 'src/components/ItemProductList'

export const Store = React.createContext<any>(null);

export interface InitialStateProps {
  loading: boolean;
  alert: string;
  products: ItemProductListProps[];
  selected: ItemProductListProps | null;
  editMode: boolean;
  openBottomSheet: boolean;
  callbackBottomSheet: () => void;
}

interface ActionProps {
  label: string;
  type: string;
  payload: any;
}

const initialState = {
  loading: true,
  alert: '',
  products: [],
  selected: null,
  editMode: false,
  openBottomSheet: false,
  callbackBottomSheet: () => null,
};

function reducer(state: InitialStateProps, action: ActionProps) {
  switch (action.type) {
    case "SET_STORE":
      return { ...state, [action.label]: action.payload };
    case "RESET_DATA":
      return { ...initialState };
    default:
      return state;
  }
}

export function StoreProvider({ children }: { children: ReactElement }) {
  const [state, dispatch] = React.useReducer(reducer, initialState);
  const value = { state, dispatch };

  return <Store.Provider value={value}>{children}</Store.Provider>;
}
