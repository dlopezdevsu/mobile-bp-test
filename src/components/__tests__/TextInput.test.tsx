/**
 * @format
 */

import 'react-native';
import React from "react";
import TextInput from "src/components/TextInput";
import renderer from "react-test-renderer";
import { StoreProvider } from 'src/hooks/main_store'

import { it, expect } from '@jest/globals';

it(`render TextInput`, () => {
  const tree = renderer.create(<StoreProvider>
    <TextInput />
  </StoreProvider>);
  expect(tree).toMatchSnapshot();
});
