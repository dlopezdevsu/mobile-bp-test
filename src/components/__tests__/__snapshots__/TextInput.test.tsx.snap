// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render TextInput 1`] = `
[
  <Text
    style={
      {
        "fontSize": 12,
        "fontWeight": "600",
        "marginBottom": 6,
        "marginTop": 18,
      }
    }
  />,
  <TextInput
    placeholderTextColor="#000000"
    style={
      [
        {
          "backgroundColor": "#ffffff",
          "borderColor": "#eaecf3",
          "borderRadius": 6,
          "borderWidth": 2,
          "padding": 12,
        },
        undefined,
        false,
        undefined,
      ]
    }
  />,
]
`;
